src/txn/txn_rollback_to_stable.c

```diff
@@ -156,6 +158,16 @@ __txn_abort_row_replace_with_hs_value(
 
     WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
 
+    /* Get the full update value from the data store. */
+    unpack = &_unpack;
+    __wt_row_leaf_value_cell(session, page, rip, NULL, unpack);
+    WT_CLEAR(ds_value);
+    if (!__wt_row_leaf_value(page, rip, &ds_value))
+        WT_ERR(__wt_page_cell_data_ref(session, page, unpack, &ds_value));
+
+    WT_CLEAR(full_value);
+    WT_ERR(__wt_buf_set(session, &full_value, ds_value.data, ds_value.size));
```

Comment:
Depending on how frequently this function is called, it might be worth using scratch buffers instead of individual allocations on every call.

Reply:
I'm not sure of the frequency - @kommiharibabu, what do you think?

Reply:
This function is called in most of the times in the recovery flow but less number of times in normal `rollback to stable` flow. I feel we can go ahead without scratch buffers and add a ticket to optimize it along with scenario where the formed update is not used because of no valid update is found.